src/react/containers/EditorContainer.js

```diff
@@ -93,6 +93,17 @@ class EditorContainer extends Component {
     const authorIds = authors.map(author => author.id);
     const author = authorIds.length > 0 ? authorIds[0] : false;
     const contributors = authorIds.length > 1 ? authorIds.slice(1, authorIds.length) : false;
+    const htmlregex = /<(img|picture|video|audio|canvas|svg|iframe|embed) ?.*>/;
+
+    // We don't want an editor publishing empty entries
```

Comment:
This functionality seems like it should be within its own function, so that the consuming code here would be as simple as (pseudo-code):

```js
// Don't allow publishing of empty entries.
if ( editorIsEmpty( editorState ) ) {
    return;
}
```

The logic is then abstracted into something like (again, pseudo-code):

```js
/**
 * Check if the content is empty.
 *
 * Look for text within the editor, the most commen case. If that's not found, look for markup indicating media / non-text content.
 *
 * @since 1.9.2
 * (Plus whatever the JS equivalent of `@param` and `@return` tags are.)
 */
editorIsEmpty( editor ) {
	const htmlregex = /<(img|picture|video|audio|canvas|svg|iframe|embed) ?.*>/;
    const content   = editor.getCurrentContent();
	
    if ( content.getPlainText().trim() ) {
        // There is text content.
        return false;
    }

    if ( htmlregex.exec(convertToHTML(content)) === null) {
        // There is some non-text content.
        return false;
    }

    return true;
}
```

(The code would be potentially more re-usable and testable, if you just pass in `editorState.getCurrentContent()`, rather than the instance of the `editorState`.)